ArduPlane/tailsitter.cpp

```diff
@@ -48,6 +103,51 @@ bool QuadPlane::tailsitter_active(void)
     return false;
 }
 
+void QuadPlane::scale_control_surfaces(float& fw_aileron, float& fw_elevator, float& fw_rudder, float& aileron, float& elevator, float& rudder){
+    float VTOL_ratio = 1.0f;
+    if (tailsitter.gain_scaling_mask & TAILSITTER_GSCL_INTERP) {
+        float aspeed;
+        bool have_airspeed = ahrs.airspeed_estimate(&aspeed);
+        const float scaling_range = tailsitter.scaling_speed_max - tailsitter.scaling_speed_min;
+        if (aspeed > tailsitter.scaling_speed_min && have_airspeed && !is_zero(scaling_range)) {
+            // apply surface scaling to interpolate between fixed wing and VTOL outputs based on airspeed
+            // kd0aij note: that the airspeed estimate based only on GPS and (estimated) wind is
+            // not sufficiently accurate for tailsitters. (based on tests in RealFlight 8 with 10kph wind)
+
+            if (!assisted_flight) {
+                // match the Q rates with plane controller
+                // no fixed wing yaw controller so cannot stabilize VTOL roll
+                const float pitch_rate = attitude_control->get_rate_pitch_pid().get_pid_info().target * 100;
+                const float yaw_rate = attitude_control->get_rate_yaw_pid().get_pid_info().target * 100;
+                const float speed_scaler = plane.get_speed_scaler();
+
+                // due to reference frame change roll and yaw are swapped, use roll as rudder input and output direct as with plane
+                fw_aileron = plane.rollController.get_rate_out(-yaw_rate, speed_scaler);
+                fw_elevator = plane.pitchController.get_rate_out(pitch_rate, speed_scaler);
+                fw_rudder = plane.channel_roll->get_control_in();
```

Comment:
I think this may not be correct with the new control options? Does the roll input always control tail-sitter roll?